.github/workflows/ci.yml

```diff
@@ -0,0 +1,79 @@
+name: CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+
+env:
+  IMAGE_TAG: ${{ github.sha }}
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - name: Build Docker images
+        run: |
+          TARGET=prod make build
+          TARGET=dev make build
+
+      - name: Save Docker images
+        run: |
+          mkdir -p docker-image-prod docker-image-dev
+          docker save -o docker-image-prod/article-page.tar libero/article-page:${IMAGE_TAG}
+          gzip docker-image-prod/article-page.tar
+          docker save -o docker-image-dev/article-page.tar libero/article-page:${IMAGE_TAG}-dev
+          gzip docker-image-dev/article-page.tar
+
+      - name: Upload prod Docker image
+        uses: actions/upload-artifact@v1
+        with:
+          name: docker-image-prod
+          path: docker-image-prod
+
+      - name: Upload dev Docker image
+        uses: actions/upload-artifact@v1
+        with:
+          name: docker-image-dev
+          path: docker-image-dev
+
+  test:
+
+    needs: build
+    runs-on: ubuntu-latest
+
+    strategy:
```

Comment:
Not sure if I fully understand what the matrix job is doing here, but would it maybe be cleaner/faster to have a single matrix job that includes build and test steps? If my limited github action knowledge isn't failing me this would parallelize the builds and avoid having to pass around container images as artifacts.

Reply:
Copied from [libero/article-store](https://github.com/libero/article-store/blob/master/.github/workflows/ci.yml), have a look through https://github.com/libero/article-store/pull/74 for context.

Improvements definitely welcome though (for both repos)!

Reply:
Hmmm after looking at the article-store I get why that one benefits from a separate build step. The various test all use the same images. Without context or measurements regarding test/build duration I think it would be premature to optimize parallelism.